cmake/SetupCodeChecks.cmake

```diff
@@ -83,15 +83,22 @@ if(CLANGQUERY_FOUND)
 endif()
 
 if(CLANGTIDY_FOUND)
+    # targets for verifying clang-tidy
     add_custom_target(clang_tidy_check)
     add_dependencies(${BLT_CODE_CHECK_TARGET_NAME} clang_tidy_check)
+
+    # targets for modifying code based via clang-tidy; also require clang-apply-replacements
+    if(CLANGAPPLYREPLACEMENTS_FOUND)
+        add_custom_target(clang_tidy_style)
+        add_dependencies(${BLT_CODE_STYLE_TARGET_NAME} clang_tidy_style)
```

Comment:
I think I should remove this line.

The changes from `clang_tidy_style` are not always safe, so it should not be added to the `style` target.
Rather, it should be opt-in, i.e. users would need to explicitly run `clang_tidy_style`.

Reply:
I think that sounds good. We should document it somewhere though, users will expect it to be under the `style` target.

Reply:
We discussed this in an axom meeting on 6/26/2023 and decided that clang-tidy targets should not be added to `make check` and `make style` by default, and that we should document this.